x-pack/plugins/observability_solution/slo/public/hooks/use_create_burn_rate_rule.tsx

```diff
@@ -78,17 +71,17 @@ export function useCreateRule<Params extends RuleTypeParams = never>() {
       },
 
       onSuccess: () => {
-        if (loadingToastId) {
-          toasts.remove(loadingToastId);
-        }
         toasts.addSuccess(
           i18n.translate('xpack.slo.rules.createRule.successNotification.descriptionText', {
             defaultMessage: 'Burn rate rule created successfully.',
           })
         );
       },
+      onSettled: (data) => {
+        if (data?.loadingToastId) {
+          toasts.remove(data?.loadingToastId);
+        }
+      },
```

Comment:
I wonder how this works locally because `loadingToastId` is in the context variable, which is the 4th argument of the `onSettled` function: https://tanstack.com/query/v4/docs/framework/react/guides/mutations#mutation-side-effects

`data` is what is returned by the mutate query function (the API call basically). 





```suggestion
      onSettled: (_, _, _, context) => {
        if (context?.loadingToastId) {
          toasts.remove(context?.loadingToastId);
        }
      },
```